modules/framework-cats/shared/src/test/scala/DogFoodTests.scala

```diff
@@ -388,19 +369,40 @@ object DogFoodTests extends IOSuite {
         expect.same(expected, actual)
     }
   }
+  test("failures in expect.all are reported with their index") {
+    _.runSuite(Meta.Clue).map {
+      case (logs, _) =>
+        val actual = extractFailureMessageForTest(logs, "(all)")
+        val expected =
+          s"""
+        |- (all) 0ms
+        | [0] assertion 2 of 5 failed: clue(x) == clue(y) (modules/framework-cats/shared/src/test/scala/Meta.scala:102)
+        | [0] 
+        | [0] Clues {
+        | [0]   x: Int = 1
+        | [0]   y: Int = 2
+        | [0] }
+        |
+        | [1] assertion 4 of 5 failed: clue(y) == clue(z) (modules/framework-cats/shared/src/test/scala/Meta.scala:102)
+        | [1] 
+        | [1] Clues {
+        | [1]   y: Int = 2
+        | [1]   z: Int = 3
+        | [1] }
```

Comment:
I wonder how useful the `2 out of 5` and `4 out of 5` are. Should we have an extra log line like:
```
assertions:
1: x == x
2: clue(x) == clue(y)
3: y == y
4: clue(y) == clue(z)
5: z == z
```
Or maybe get rid of the numbering altogether?

Reply:
Getting rid of the numbering is easiest to maintain. I added the numbering at first, thinking the source code would be challenging.

We can work on formatting source code in error messages in future PRs.